dashboard/test/models/sections/section_test.rb

```diff
@@ -478,4 +478,90 @@ def verify(actual, expected)
     refute Section.valid_grade?("Something else")
     refute Section.valid_grade?("56")
   end
+
+  class HasSufficientDiscountCodeProgress < ActiveSupport::TestCase
+    self.use_transactional_test_case = true
+
+    def create_script_with_levels(name, level_type)
+      script = create :script, name: name
+      stage = create :stage, script: script
+      # 10 non-programming levels
+      5.times do
```

Comment:
Misleading comment - looks like we actually create 5 non-programming levels?  And below.